pages/intro.js

```diff
@@ -0,0 +1,197 @@
+        e.preventDefault();
+        calculateTextAreaHeight()
+        if (type == 'intro') {
+            setIsEditingIntro(!isEditingIntro)
+        } else {
+            setIsEditingOthers(!isEditingOthers)
+        }
+    };
+
+    const handleCancel = (type, e) => {
+        e.preventDefault();
+        if (type == 'intro') {
+            setIsEditingIntro(false)
+        } else {
+            setIsEditingOthers(false)
+        }
+    }
+
+    const handleSave = (e) => {
+        // TODO: update firebase with value of div
+        e.preventDefault();
+        setIsEditing(false)
+    }
+
+    // sets the heights for all textareas based on their scroll height
+    const calculateTextAreaHeight = () => {
+        const textareas = document.getElementsByClassName('textarea');
+        Array.prototype.forEach.call(textareas, function (textarea) {
+            textarea.style.height = textarea.scrollHeight + 'px'
+        })
+    }
+
+    useEffect(() => {
+        calculateTextAreaHeight()
+    })
+
+    function Content(props) {
+        const { header, content, isEditing, cancel } = props
+        return (
+            <React.Fragment>
+                {
+                    isEditing ?
+                        <React.Fragment>
+                            <StyledHeader>Header</StyledHeader>
+                            <EditingHeader className='textarea' defaultValue={header} />
+                            <StyledHeader>Body</StyledHeader>
+                            <EditingText className='textarea' defaultValue={content} />
+                            <div style={{ height: '27px', backgroundColor: `${COLOR.BACKGROUND}` }} />
+                            <div style={{ display: 'flex' }}>
+                                <StyledCancel onClick={cancel}>
+                                    <p style={{ borderBottom: `2px solid ${COLOR.BLACK}`, margin: '0px' }}>
+                                        Cancel
+                                    </p>
+                                </StyledCancel>
+                                <StyledSave>Save</StyledSave>
+                            </div>
+                        </React.Fragment>
+                        :
+                        <React.Fragment>
+                            <HeaderText className='textarea' disabled defaultValue={header} />
+                            <ContentText className='textarea' disabled defaultValue={content} />
+                        </React.Fragment>
+                }
+            </React.Fragment>
+        )
+    }
+
+    return (
+        <React.Fragment>
+            <div style={{ marginTop: '-40px', marginBottom: '40px' }}>
+                <Card>
+                    <CardHeader>
+                        <CardTitle>Intro Text</CardTitle>
+                        <p>{props.nwHacks.subtitle}</p>
+                        <CardButtonContainer>
+                            <Button type={EDIT} onClick={handleEdit.bind(null, 'intro')} />
```

Comment:
does this work? Slightly more modern syntax
```suggestion
                            <Button type={EDIT} onClick={() => handleEdit('intro')} />
```